modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/gpi/GPIClient.scala

```diff
@@ -153,18 +161,20 @@ object GPIExample extends App {
 
   import cats.effect.IO
   import scala.concurrent.duration._
+  import cats.instances.string._
 
   private val gpiStatus =
     Stream.bracket(
       Giapi.giapiConnection[IO]("failover:(tcp://127.0.0.1:61616)", 2000.millis).connect)(
       giapi => {
-        val client = new GPIClient[IO](giapi)
+        val client = new GPIClient[IO](giapi, scala.concurrent.ExecutionContext.Implicits.global)
         val r =
           for {
-            h <- client.heartbeat
-            f <- client.fpmMask
-            o <- client.aoDarkLevel
-          } yield (h, f, o)
+            hs <- client.heartbeatS.flatMap(_.take(3).compile.toVector)
+            h  <- client.heartbeat
+            f  <- client.fpmMask
+            o  <- client.aoDarkLevel
+          } yield (hs, h, f, o)
```

Comment:
On running this against GPI it produces:
`(Vector(5738574, 5738575, 5738576),5738577,FPM_H,-8.0)`

Which corresponds to three heartbeat items, then the other variables